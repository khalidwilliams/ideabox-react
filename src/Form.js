import React from 'react';

const Form = ({addCard}) => {
  return (
    <form>
      <div className="label-group">
        <label for="title-input">Idea Title</label>
        <input id="title-input" type="text" />
      </div>
      <div className="label-group">
        <label for="description-input">Idea Title</label>
        <input id="description-input" type="text"/>
      </div>
      <button className="add-card-btn" onClick={() => addCard()}>Add Idea</button>
    </form>
  );
};

export default Form;
