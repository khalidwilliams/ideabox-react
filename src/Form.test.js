import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form.js';

describe('Form Component', () => {
  let addCardMock;
  beforeEach(() => {
    addCardMock = jest.fn();
  });

  it('should render a form', () => {
    const shallowForm = shallow(<Form 
        addCard={addCardMock}
      />);

    expect(shallowForm).toMatchSnapshot();
  });

  it('should add a card', () => {
    const shallowForm = shallow(<Form 
        addCard={addCardMock}
      />);
    shallowForm.find('.add-card-btn').simulate('click');
    expect(addCardMock).toHaveBeenCalledTimes(1);
  });
  
  
});
