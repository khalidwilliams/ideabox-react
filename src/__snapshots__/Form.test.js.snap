// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should render a form 1`] = `
<form>
  <div
    className="label-group"
  >
    <label
      for="title-input"
    >
      Idea Title
    </label>
    <input
      id="title-input"
      type="text"
    />
  </div>
  <div
    className="label-group"
  >
    <label
      for="description-input"
    >
      Idea Title
    </label>
    <input
      id="description-input"
      type="text"
    />
  </div>
  <button
    className="add-card-btn"
    onClick={[Function]}
  >
    Add Idea
  </button>
</form>
`;
